build/start.sh

```diff
@@ -333,89 +358,83 @@ if [ -n "$verbose" ]; then
-  # Build KairosDb Docker Image (with KairosDb-Extensions)
-  # TODO(ville): Build this from source instead of using released version
-  # https://github.com/ArpNetworking/metrics/issues/30
-  #echo "Building kairosdb-extensions..."
+if [ $start_kairosdb -gt 0 ] && [ -z "$skip_build" ]; then
+  pushd ${dir_kairosdb} &> /dev/null
+  printf "Building kairosdb...\n"
+  ./jdk-wrapper.sh ./mvnw package -DskipAllVerification=true -DskipSources=true -DskipJavaDoc=true
+  if [ "$?" -ne 0 ]; then printf "\e[0;31mError\e[0m: Build failed of kairosdb\n"; exit 1; fi
+  popd &> /dev/null
 fi
 
 if [ $start_metrics_portal -gt 0 ] && [ -z "$skip_build" ]; then
   pushd ${dir_metrics_portal} &> /dev/null
-  echo "Building metrics-portal..."
+  printf "Building metrics-portal...\n"
   ./jdk-wrapper.sh ./mvnw package -DskipAllVerification=true -DskipSources=true -DskipJavaDoc=true
-  if [ "$?" -ne 0 ]; then echo "Build failed: metrics-portal"; exit 1; fi
+  if [ "$?" -ne 0 ]; then printf "\e[0;31mError\e[0m: Build failed of metrics-portal\n"; exit 1; fi
   popd &> /dev/null
 fi
 
 if [ $start_mad -gt 0 ] && [ -z "$skip_build" ]; then
   pushd ${dir_mad} &> /dev/null
-  echo "Building metrics-aggregator-daemon..."
+  printf "Building metrics-aggregator-daemon...\n"
   ./jdk-wrapper.sh ./mvnw package -DskipAllVerification=true -DskipSources=true -DskipJavaDoc=true
-  if [ "$?" -ne 0 ]; then echo "Build failed: metrics-aggregator-daemon"; exit 1; fi
+  if [ "$?" -ne 0 ]; then printf "\e[0;31mError\e[0m: Build failed of metrics-aggregator-daemon\n"; exit 1; fi
   popd &> /dev/null
 fi
 
 if [ $start_cluster_agg -gt 0 ] && [ -z "$skip_build" ]; then
   pushd ${dir_cluster_agg} &> /dev/null
-  echo "Building metrics-cluster-aggregator..."
+  printf "Building metrics-cluster-aggregator...\n"
   ./jdk-wrapper.sh ./mvnw package -DskipAllVerification=true -DskipSources=true -DskipJavaDoc=true
-  if [ "$?" -ne 0 ]; then echo "Build failed: cluster-aggregator"; exit 1; fi
+  if [ "$?" -ne 0 ]; then printf "\e[0;31mError\e[0m: Build failed of cluster-aggregator\n"; exit 1; fi
   popd &> /dev/null
 fi
 
 # Run projects
-if [ ${start_ckg} -gt 0 ]; then
-  rm -f "${target}/kairosdb.properties"
-
+if [ ${start_cg} -gt 0 ]; then
   # Cassandra
   pid_cassandra=$(docker run -d -p 7000:7000 -p 9042:9042 cassandra:${docker_cassandra_version})
-  echo "Started: cassandra (${pid_cassandra:0:12})"
-
-  # TODO(ville): We need to wait for cassandra; otherwise kairosdb just exits
-  # https://github.com/InscopeMetrics/kairosdb-extensions/issues/26
-  cassandra_healthy=0
-  while [ ${cassandra_healthy} -eq 0 ]
-  do
-    echo "Waiting for cassandra to become healthy..."
-    sleep 2
-    if docker exec ${pid_cassandra} cqlsh &> /dev/null ; then
-      cassandra_healthy=1
-    fi
-  done
-  echo "Cassandra is healthy!"
-
-  if [ -n "${pinger}" ]; then
-    "${dir}/pinger.sh" ${verbose_arg} -c "${pid_cassandra}                             " -n "cassandra" -p "${pid_cassandra:0:12}" &
-  fi
-
```

Comment:
I see that this issue isn't closed, but I'll trust that this is no longer necessary and was removed on purpose.

Reply:
This PR does not address https://github.com/InscopeMetrics/kairosdb-extensions/issues/26 per say but works around it by telling Docker to continually restart the container if it exits without being explicitly stopped. The issue remains open, because in the wild we would prefer the KDB container to wait patiently for Cassandra to become available instead of thrashing the way it does (or least I think that would be an improvement).

Reply:
(Hence 26 will not be closed with this PR)